roqoqo-iqm/src/interface.rs

```diff
@@ -70,26 +75,37 @@ pub struct IqmInstruction {
 /// `device_number_qubits` - The number of qubits of the backend device. It is used to know how many
 ///  qubits to measure with [roqoqo::operations::PragmaRepeatedMeasurement]
 ///
+/// `output_registers` - A mutable reference to the classical registers that need to be initialized
+///
 /// # Returns
 ///
-/// `Ok(IqmCircuit)` - Converted circuit `Err(RoqoqoBackendError::OperationNotInBackend)` - Error
-/// when [roqoqo::operations::Operation] can not be converted
+/// `Ok(IqmCircuit, RegisterMapping, usize)` - Converted circuit, mapping of measured qubits to register indices, and number of measurements
+/// `Err(RoqoqoBackendError::OperationNotInBackend)` - Error when [roqoqo::operations::Operation] can not be converted
 pub fn call_circuit<'a>(
     circuit: impl Iterator<Item = &'a Operation>,
     device_number_qubits: usize,
-) -> Result<(IqmCircuit, usize), RoqoqoBackendError> {
+    output_registers: &mut HashMap<String, BitOutputRegister>,
+) -> Result<(IqmCircuit, RegisterMapping, usize), RoqoqoBackendError> {
     let mut circuit_vec: Vec<IqmInstruction> = Vec::new();
     let mut number_measurements: usize = 1;
+    let mut register_mapping: RegisterMapping = HashMap::new();
 
     for op in circuit {
         match op {
-            // Do nothing with definitions of registers, since the name of the output register will
-            // be contained in the `args` field of the measurement instruction
-            Operation::DefinitionBit(_) => {}
-            Operation::DefinitionFloat(_) => {}
-            Operation::DefinitionComplex(_) => {}
```

Comment:
Are these types no longer allowed?

Reply:
Discussing with @nfwvogt I recall him saying that we only care about the BitRegisters in the output. Not sure if I should just ignore these other two if they are present in the circuit, or raise an OperationNotInBackend error.